Custom Variant Settings, Player representation, Events and Events, and a lot of refactoring

Kantanomo opened the pull request:
Contains all of the Custom variant setting code and playlist loader hooks
Custom Settings
- gravity,
- infinite_ammo,
- explosion_physics,
- hill_rotation,
- game_speed,
- infinite_grenades,
- spawn_protection,
- hill_set,
- forced_fov

Added hooks for adding new player_represntation types for Multiplayer only, fallback code has been put in for clients/servers that fail to load the new types to prevent crashes.

New spooky event

Moved around and cleaned up a lot of code, and added a handful of new structure definitions.


Completely rewrote the EventHandler to be more extendable and easier to use.

Kantanomo committed:
Added some comments

Kantanomo committed:
Migration and changes

Implemented Engine calls for get_current_engine_type and get_global_random_seed_address

Migrated CustomVariantSettings and PlaylistLoader from old branch

Added struct for physics_constants (Incomplete)

Added new event handler using a new format for countdown start,  (Need to change all existing event handlers to use it.)

Moved King of the Hill rotation hook to CustomVariantSettings

Added Hill Rotation as a custom setting for playlists

Kantanomo committed:
Simplifying Playlist loader and EventHandler

Simplified Playlist loader, no need to hook the whole ReadLine function just the ProcessProperty method for custom settings.

Started the ground work to remake EventHandler to be a lot easier to use.

Kantanomo committed:
Events. Events. Events.

More work on the new implementation of events

Kantanomo committed:
Event Handler Changes

Replaced all event handlers to use the new system, most tested but still doing more testing.

Kantanomo committed:
Merge pull request #17 from pnill/development-new

Pull from Origin

Kantanomo committed:
Merge branch 'PlaylistExtensionNew' into PlaylistMergeTemp

Kantanomo committed:
Merge pull request #18 from Kantanomo/PlaylistMergeTemp

Moving over base changes.

Kantanomo committed:
Some little changes

Kantanomo committed:
Lots of Changes

Moved GAME_BUILD and EXECUTABLE_VERSION to defines inside of H2MOD.h

Renamed Engine:: to EngineCalls::

Created EngineHooks:: to be utilized for any hooks that are more of a global thing, part of the H2MOD.cpp cleanup

Added a function to PhysicsConstants to get the default gravity value.

Change Custom Setting Gravity to be a ratio of default gravity (Setting 1.0 = Default Gravity)

Added Gamespeed and Infinite Grenades as a setting.

Added Bluescreen detection and Player Spawn/Death Events.

Changed some Types in GunGame to use Datum instead of Int

Added a little bit more logging to StatsHandler

Kantanomo committed:
Small commit to switch branches.

Kantanomo committed:
Blam Def Changes, Cleanup, and Custom Settings

Updated some blam def structs and enums to be more descriptive of what they actually represent.

Cleaned up some typedefs so there is less clutter.

Added a predefined settings for hill rotation to manually define which hills are used in rotation.

Added a setting for controling spawn protection time.

added a util functions for testing if a string is an integer.

re-added mapload event execution.

Kantanomo committed:
Merge remote-tracking branch 'upstream/development-new' into PlaylistExtensionNew

Kantanomo committed:
Some fixes so it compiles

Kantanomo committed:
These are variant settings... not playlist settings

oops.

Kantanomo committed:
Def Changes and experimental code.

Kantanomo committed:
Merge pull request #20 from Kantanomo/PlaylistExtensionNew

Playlist extension new

Kantanomo committed:
refactoring and lazy_blam

moved some structs into a more global namespace.

added some missing operators to real_math types

started implimentation of lazy_blam for loading tag data  from a cache file.

Kantanomo committed:
uneeded code.

Kantanomo committed:
test

Kantanomo committed:
lazy blam updates

Kantanomo committed:
Lazy Blam updates

Finishing biped loading, everything seems to check out on rebasing and pulling all tag_block data.

Maybe went a little macro crazy but it cleans things up

Kantanomo committed:
lazy_blam updates

Kantanomo committed:
Lazy Blam Rebase work

What in the world even is this.

Kantanomo committed:
Lazy Blam Changes

Switched to using for-each looping methods and finished globals

Kantanomo committed:
Lazy Blam Updates

Implemented a LazyBuffer to help with loading tag data, it expands itself to be larger while freeing old buffers.

Discovered that the instance size does not include things outside of the scope of the tag_data (i.e datarefs, opmitized tag blocks) which makes the LazyBuffer essential for loading them.

Kantanomo committed:
Lazy Blam Updates

I love macros.

Kantanomo committed:
Commit before pulling master into branch.

Kantanomo committed:
Merge remote-tracking branch 'upstream/development-new' into development-new

Kantanomo committed:
Some things that slipped through the cracks pulling changes over.

Kantanomo committed:
Changes and Player representation testing

Kantanomo committed:
tiny bit of reorganization

Kantanomo committed:
Merge remote-tracking branch 'upstream/development-new' into development-new

Kantanomo committed:
wew lad

Kantanomo committed:
Merge remote-tracking branch 'upstream/development-new' into development-new

Kantanomo committed:
Merge remote-tracking branch 'upstream/development-new' into development-new

Remove some useless includes

Himanshu-01 committed:
Rename Datum Macros to standard known names

Himanshu-01 committed:
Update s_object_placement_data

Himanshu-01 committed:
Update Blam Objects

Himanshu-01 committed:
Update s_game_variant
-Rename s_game_engine_settings ->s_game_options

Himanshu-01 committed:
Update s_game_globals

Himanshu-01 committed:
Add s_game_state_header

Himanshu-01 committed:
Code Cleanup
-Used IncludeToolBox to refactor Includes in most files

Kantanomo committed:
Merge pull request #22 from Himanshu-01/objects

Some Blam stuffs

Himanshu-01 committed:
More Blam Updates
-Start adding Simulation and Networking stuffs
-refactored PlayerControls

Himanshu-01 committed:
Add structures for unit variant sync

Kantanomo committed:
Merge pull request #23 from Himanshu-01/objects

Adds some simulation structures

Kantanomo committed:
Changes

Fixed the new PlayerControl changes.

Added setting to choose to play spooky boy only during the event times (Currently always forced during development)

Kantanomo committed:
More changes

Added server offsets that were missing for memory calls

Made it so outside of the Halloween event you can't use the skeleton biped

fixed an issue with the tag_loader resolving to tags loaded in previous game sessions

Added some more pumpkins

Bertha wouldn't stop complaining so Skeleton is forced on by default for the event.

nukeulater commented on xlive/Blam/Cache/DataTypes/BlamTag.h:
```diff
@@ -127,6 +127,7 @@ struct blam_tag
 		soundcachefilegestalt = 'ugh!',
 		cachefilesound = '$#!+',
 		mousecursordefinition = 'mcsr',
+		uldg = 'udlg'
```
how is it in the end? udlg or uldg

NukeULater commented:
pre-compiled header should be included first, please move #include "stdafx.h" back or just straight up remove the include

nukeulater commented on xlive/H2MOD/Modules/PlaylistLoader/PlaylistLoader.cpp:
```diff
@@ -0,0 +1,302 @@
+				return custom_setting.second;
+		return none;
+	}
+	void playlist_invalid_item_hook(playlist_entry* playlist_entry, int a2, int a3, int a4, wchar_t *a5, wchar_t *a6, wchar_t *a7)
```
this looks very ugly

Kantanomo replied:
Which part, cause it's all bad.

nukeulater replied:
Lol

Kantanomo replied:
Tried the way that you suggested and it does not work, it just crashes right away.

Kantanomo replied:
fixed

nukeulater commented on xlive/H2MOD/Modules/SpecialEvents/SpecialEvents.cpp:
```diff
@@ -279,7 +310,384 @@ namespace SpecialEvents
+		{1, 0, -2.072325f, 5.767787f, -1.4f, 0.f, 0.f, 0.f, 1.f},
+		{1, 0, -0.955f, 5.767787f, -1.4f, 0.f, 0.f, 0.f, 1.f}
+	};
+	std::vector<std::tuple<byte, byte, float, float, float, float, float, float, float>> coag_scen_places
```
probably vectors, no need for all those floats there

Kantanomo replied:
I don't understand.

nukeulater replied:
Use real_vector3d for example instead of 3 floats

Kantanomo replied:
Fixed, please for my sanity never ask me to do that again 😂

Kantanomo committed:
Some fixes

fixed a tag enum not being named correctly

fixed infection not allowing skeleton bipeds

Kantanomo committed:
DX9 Libs

Kantanomo committed:
But why though

Please never ask me to do a change like this ever again.

Co-Authored-By: NukeULater <18323679+NukeULater@users.noreply.github.com>

Kantanomo committed:
17th times the charm

Finally got rid of one of the ugly calls.

nukeulater commented on xlive/H2MOD/Modules/SpecialEvents/SpecialEvents.cpp:
```diff
@@ -522,15 +522,12 @@ namespace SpecialEvents
-					ori.z = std::get<7>(scen_place);
-					set_orientation(&placement.orientation, &placement.up, &ori);
-					placement.scale = std::get<8>(scen_place);
+					placement.position.x = std::get<2>(scen_place).i;
```
u can just placement.position = std::get<2>(scen_place);, no need to go through all components

Kantanomo replied:
it places them in i,j,k instead of x,y,z when using those types. so everything gets placed at 0,0,0

Kantanomo committed:
Missed a call to swap out

Kantanomo committed:
Mixed up operators

Typo in the equals operator for real_vector3d

nukeulater commented on xlive/Blam/Math/real_math.h:
```diff
@@ -24,6 +24,17 @@ struct angle
 		rad(_rad)
 	{}
 
+	bool operator=(const angle& other) const
```
u forgot this

Kantanomo replied:
fixed

Kantanomo committed:
Another one

Kantanomo committed:
Remove forced SpecialEvent.

Kantanomo committed:
Revert "DX9 Libs"

This reverts commit bdd26e317d8a8858e5b710b1641b26925a2753c9.

Kantanomo committed:
Data fix

Fixed something that should be setting a value but was a comparison, it's not necessary to set it but might aswell stay in line with the expected struct conventions.

Kantanomo committed:
disabling my testing class
